sentinelsat/sentinel.py

```diff
@@ -289,6 +293,53 @@ def get_footprints(self, products):
             )
         return geojson.FeatureCollection(feature_list)
 
+    def to_dict(self, products):
+        """Return the products in a dictionary object with the values in their appropriate Python types"""
+        try:
+            strtype = basestring
+        except:
+            strtype = str
+        output = {}
+        for prod in products:
+            prodname = prod['title']
+            output[prodname] = {}
+            for key in prod:
+                if key != 'title' and isinstance(prod[key], strtype):
+                    output[prodname][key] = prod[key]
+                elif isinstance(prod[key], dict):
+                    output[prodname][prod[key]['name']] = prod[key]['content']
+                elif key == 'link':
+                    for link in prod[key]:
+                        if not link.has_key('rel'):
+                            output[prodname]['link'] = link['href']
+                        else:
+                            lkey = 'link_' + link['rel']
+                            output[prodname][lkey] = link['href']
+                elif key == 'date':
+                    for data in prod[key]:
+                        mydate = None
+                        try:
+                            mydate = datetime.strptime(data['content'],'%Y-%m-%dT%H:%M:%SZ')
+                        except:
+                            try:
+                                mydate = datetime.strptime(data['content'],'%Y-%m-%dT%H:%M:%S.%fZ')
+                            except:
+                                print("Date '{dat}' non parsable".format(dat=data['content']))
```

Comment:
This should be an exception, not a print statement. Maybe a new Exception `class DatetimeParsingError(Exception`. (see for instance the `InvalidChecksumError`).

see also #83